components/cluster.py

```diff
@@ -0,0 +1,195 @@
+import numpy as np 
+import pandas as pd 
+import dash_html_components as html
+import dash_core_components as dcc
+from dash.dependencies import Input, Output, State
+import dash
+import dash_leaflet as dl
+
+from app import app
+import datetime 
+from utils import earthquake_data
+import math
+from numba import njit, jit
+import plotly.graph_objects as go
+import networkx as nx
+from datetime import datetime as dt
+
+
+
+def get_data( session_id ):
+    eq_data = earthquake_data.get_earthquake_data(session_id)
+    return eq_data
+def compute_edges( df ):
+    # add needed columns
+    if df.empty: 
+        return []
+    #df['DateTime'] = df.apply(lambda x: get_datetime(x), axis = 1)
+    #df["TimeStamp"] = df.DateTime.values.astype(np.int64) / 10 ** 9
+
+    data = df[[ "TimeStamp" , "EVENTID" , "TimeStamp" , "MAGNITUDE" , "LATITUDE" , "LONGITUDE" ] ].values
+    edges = compute_edges_numba( data )
+    return edges
+
+
+@jit( nopython=True )
```

Comment:
What does this do?

Reply:
So as I understand it, this jits the method to lower-level code that's more efficient (has to do with numba).